src/Repository/Storable/Introduction.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Friendica\Repository\Storable;
+
+use Friendica\Collection;
+use Friendica\Core\Protocol;
+use Friendica\Model;
+use Friendica\Network\HTTPException;
+use Friendica\Protocol\ActivityPub;
+use Friendica\Protocol\Diaspora;
+use Friendica\Repository\Storable;
+use Friendica\Util\DateTimeFormat;
+
+/**
+ * @method Model\Introduction       selectFirst(array $condition)
+ * @method Collection\Introductions select(array $condition = [], array $params = [])
+ * @method Collection\Introductions selectByBoundaries(array $condition = [], array $params = [], int $max_id = null, int $since_id = null, int $limit = self::LIMIT)
+ */
+class Introduction extends Storable
+{
+	protected static $table_name = 'intro';
+
+	protected static $model_class = Model\Introduction::class;
+
+	protected static $collection_class = Collection\Introductions::class;
+
+	/**
+	 * Confirms a follow request and sends a notic to the remote contact.
+	 *
+	 * @param Model\Introduction $Introduction
+	 * @param bool               $duplex       Is it a follow back?
+	 * @param bool|null          $hidden       Should this contact be hidden? null = no change
+	 * @return bool
+	 * @throws HTTPException\InternalServerErrorException
+	 * @throws HTTPException\NotFoundException
+	 * @throws \ImagickException
+	 */
+	public function confirm(Model\Introduction $Introduction, bool $duplex = false, bool $hidden = null)
```

Comment:
puh .. this one goes a little bit more further than "just" repository logic. I must confess I don't feel comfortable with it .. Repositories are mainly for "plain" create/update/delete/select(specific). This one calls a lot more logic and dependencies.. I'd like to have it inside `src/Model/` again. Here we would have the namespace for "business logic"..

Reply:
But you said Repositories were for business logic? I'm utterly confused now.